source/common/network/utility.cc

```diff
@@ -342,5 +346,28 @@ bool Utility::portInRangeList(const Address::Instance& address, const std::list<
   return false;
 }
 
+absl::uint128 Utility::Ip6ntohl(const absl::uint128& address) {
+  // TODO(ccaraman): Support Ip6ntohl for big-endian.
+  ASSERT(ABSL_IS_LITTLE_ENDIAN);
```

Comment:
I would make both this assert and the other similar ones static_assert. We should just prevent the code from compiling on big endian systems until we implement this. With static_assert, the compilation message will be very clear also, much like `#error`